### PR TITLE
[SPARK-47928][SQL][TEST] Speed up test "Add jar support Ivy URI in SQL"

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

`SQLQuerySuite`/"SPARK-33084: Add jar support Ivy URI in SQL" uses Hive deps to test `ADD JAR` which pulls tons of transitive deps, this PR replaces it with light jars but covers all semantics to speed up the UT.

### Why are the changes needed?

Speed up the test, and reduce unnecessary relationships with Hive.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Run UT locally.

Before
```
[info] - SPARK-33084: Add jar support Ivy URI in SQL (16 minutes, 55 seconds)
```
After
```
[info] - SPARK-33084: Add jar support Ivy URI in SQL (17 seconds, 783 milliseconds)
```
### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala [hunk 1]
```diff
@@ -3748,22 +3748,21 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
...
     // default transitive=true, test download ivy URL jar return multiple jars
-    sql("ADD JAR ivy://org.scala-js:scalajs-test-interface_2.12:1.2.0")
-    assert(sc.listJars().exists(_.contains("scalajs-library_2.12")))
-    assert(sc.listJars().exists(_.contains("scalajs-test-interface_2.12")))
-
-    sql(s"ADD JAR ivy://org.apache.hive:hive-contrib:$hiveVersion" +
-      "?exclude=org.pentaho:pentaho-aggdesigner-algorithm&transitive=true")
-    assert(sc.listJars().exists(_.contains(s"org.apache.hive_hive-contrib-$hiveVersion.jar")))
-    assert(sc.listJars().exists(_.contains(s"org.apache.hive_hive-exec-$hiveVersion.jar")))
-    assert(!sc.listJars().exists(_.contains("org.pentaho.pentaho_aggdesigner-algorithm")))
```
Comment (pan3793) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala:3766]: this assertion is invalid. should be 
```patch
- org.pentaho.pentaho_aggdesigner-algorithm
+ org.pentaho_pentaho-aggdesigner-algorithm
```

**Comment by pan3793** [comment]
cc @dongjoon-hyun @LuciferYang

**Review by LuciferYang — COMMENTED** [review]
acceptable to me

**Comment by pan3793** [comment]
cc @yaooqinn @AngersZhuuuu

**Review by yaooqinn — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by LuciferYang** [comment]
Merged into master. Thanks @pan3793 and @yaooqinn

**Review by dongjoon-hyun — COMMENTED** [review]
Hi, @pan3793 , @LuciferYang , @yaooqinn .

Although I understand the intention, this PR looks like a regression because it removes Hive dependency test coverage. :)

**Comment by dongjoon-hyun** [comment]
Does this PR aim to mitigate this?
- https://github.com/apache/spark/pull/45372#discussion_r1574143381

**Comment by yaooqinn** [comment]
Hi @dongjoon-hyun, tests of HiveVersionsSuite might cover hive dependency

**Pull request closed without merging**